bin/plugin/commands/translations.js

```diff
@@ -0,0 +1,169 @@
+ */
+
+/**
+ * @typedef WPTranslationsSettings
+ *
+ * @property {string} textdomain Plugin textdomain.
+ * @property {string} directory  Modules directory.
+ * @property {string} output     Output PHP file.
+ */
+
+const options = [
+	{
+		argname: '-d, --directory <directory>',
+		description: 'Modules directory',
+	},
+	{
+		argname: '-d, --output <output>',
+		description: 'Output file',
+	},
+];
+
+/**
+ * Command that generates a PHP file from module header translation strings.
+ *
+ * @param {WPTranslationsCommandOptions} opt
+ */
+async function handler( opt ) {
+	await createTranslations( {
+		textdomain: config.textdomain,
+		directory: opt.directory || 'modules',
+		output: opt.output || 'module-i18n.php',
+	} );
+}
+
+module.exports = {
+	options,
+	handler,
+};
+
+const TAB = '\t';
+const NEWLINE = '\n';
+const FILE_HEADER =
+	[
+		'<?php',
+		'/* THIS IS A GENERATED FILE. DO NOT EDIT DIRECTLY. */',
+		'$generated_i18n_strings = array(',
+	].join( NEWLINE ) + NEWLINE;
+const FILE_FOOTER =
+	NEWLINE +
+	[ ');', '/* THIS IS THE END OF THE GENERATED FILE */' ].join( NEWLINE ) +
+	NEWLINE;
+
+/**
+ * Parses module header translation strings.
+ *
+ * @param {WPTranslationsSettings} settings Translations settings.
+ *
+ * @return {[]string} List of translation strings.
+ */
+async function getTranslations( settings ) {
+	const moduleFilePattern = path.join( settings.directory, '*/load.php' );
+	const moduleFiles = await glob( path.resolve( '.', moduleFilePattern ) );
+
+	const moduleTranslations = await Promise.all(
+		moduleFiles.map( async ( moduleFile ) => {
+			const fileStream = fs.createReadStream( moduleFile );
+
+			const rl = readline.createInterface( {
+				input: fileStream,
+			} );
+
+			const headers = [ 'Module Name', 'Description' ];
+			const translationStrings = [];
+
+			for await ( const line of rl ) {
+				headers.forEach( ( header, index ) => {
+					const regex = new RegExp(
+						'^(?:[ \t]*<?php)?[ \t/*#@]*' + header + ':(.*)$',
+						'i'
+					);
+					const match = regex.exec( line );
+
+					if ( match ) {
+						const headerValue = match[ 1 ]
+							.replace( /\s*(?:\*\/|\?>).*/, '' )
+							.trim();
+						if ( headerValue ) {
+							headers.splice( index, 1 );
+							translationStrings.push( headerValue );
+						}
+					}
+				} );
+			}
+
+			return translationStrings;
+		} )
+	);
+
+	return flatten( moduleTranslations );
+}
+
+/**
+ * Parses module header translation strings.
+ *
+ * @param {[]string}               translations List of translation strings.
+ * @param {WPTranslationsSettings} settings     Translations settings.
+ */
+function createTranslationsPHPFile( translations, settings ) {
+	const output = translations.map( ( translation ) => {
+		// Escape single quotes.
+		return (
+			TAB +
+			`__( '${ translation.replace( /'/g, "\\'" ) }', '${
+				settings.textdomain
+			}' ),`
+		);
+	} );
+
+	const fileOutput = FILE_HEADER + output.join( NEWLINE ) + FILE_FOOTER;
```

Comment:
Using template literals.

```suggestion
	const fileOutput = `${FILE_HEADER}${ NEWLINE }${FILE_FOOTER}`;
```

Reply:
Updated (keeping the `output.join( NEWLINE )` though, the suggestion here is slightly off in that regard :)